filesystem/apps/rusthello/src/interface.rs

```diff
@@ -0,0 +1,299 @@
+There are 64 identical pieces called disks, which are white on one side and black on the other. \
+A player is Light, using disks’ white side, and the other one is Dark, using disks' black side. \
+The game starts with four disks already placed at the centre of the board, two for each side. \
+Dark moves first.\n
+Let’s say it’s Dark’s turn, for simplicity's sake, as for Light the rules are just the same. \
+Dark has to place a disk in a free square of the board, with the black side facing up. \
+Whenever the newly placed black disk and any other previously placed black disk enclose a sequence of white disks (horizontal, vertical or diagonal and of any length), all of those flip and turn black. \
+It is mandatory to place the new disk such that at least a white disk is flipped, otherwise the move is not valid.\n
+Usually players’ turn alternate, passing from one to the other. \
+When a player cannot play any legal move, the turn goes back to the other player, thus allowing the same player to play consecutive turns. \
+When neither player can play a legal move, the game ends. \
+In particular, this is true whenever the board has been completely filled up (for a total of 60 moves), but games happen sometimes to end before that, leaving empty squares on the board.\n
+When the game ends, the player with more disks turned to its side wins. \
+Ties are possible as well, if both player have the same number of disks.\n\n\n
+\tHOW TO USE RUSThello:\n
+To play RUSThello you first have to choose who is playing on each side, Dark and Light. \
+You can choose a human players or an AI. \
+Choose human for both players and challenge a friend, or test your skills against an AI, or even relax and watch as two AIs compete with each other: all matches are possible!\n
+As a human player, you move by entering the coordinates (a letter and a number) of the square you want to place your disk on, e.g. all of 'c4', 'C4', '4c' and '4C' are valid and equivalent coordinates. \
+For your ease of use, all legal moves are marked on the board with a *.\n
+Furthermore, on your turn you can also input special commands: 'undo' to undo your last move (and yes, you can 'undo' as many times as you like) and 'quit' to quit the game.\n\n\n
+\tCREDITS:\n
+RUSThello v. 1.1.0 Redox Edition
+by Enrico Ghiorzi, with the invaluable help of the Redox community
+Copyright (c) 2015 by Enrico Ghiorzi
+Released under the MIT license\n\n\n";
+
+
+pub fn input_main_menu() -> UserCommand {
+
+	loop {
+		print!("Insert input: ");
+		match get_user_command() {
+			Some(UserCommand::NewGame)	=> return UserCommand::NewGame,
+			Some(UserCommand::Help) 	=> return UserCommand::Help,
+			Some(UserCommand::Quit) 	=> {
+				println!("\nGoodbye!\n\n\n");
+				return UserCommand::Quit;
+			}
+			_ => println!("This is not a valid command!"),
+		}
+	}
+}
+
+pub fn new_player(side: reversi::Disk) -> Option<players::Player> {
+	loop {
```

Comment:
Sorry @Ticki I think I totally misunderstood your idea. I can implement an iterator like this
```
impl Iterator for UserCommand {
	type Item = UserCommand;

	fn next(&mut self) -> Option<UserCommand> {
            \\ returns a UserCommand matching user's input
        }
    }
}
```
but it seems wrong to me: it's not `UserCommand` which should produce a next `UserCommand` (and indeed `&mut self` is totally useless). Moreover I can't see a natural way to write the for-loop with this iterator.